registry/quilt_server/views.py

```diff
@@ -1620,6 +1623,24 @@ def tag_put(owner, package_name, package_tag):
     if instance is None:
         raise ApiException(requests.codes.not_found, "Package hash does not exist")
 
+    # Check to see if the tag update is to a team-shared package
+    # and therefore allowed
+    if g.auth.user != owner:
+        # Check if the package is shared team-wide
+        team_access = (
+                Access.query
+                .filter(sa.and_(
+                    Access.package == instance.package,
+                    Access.user == TEAM
+                ))
+                .one_or_none()
+            )
+        if not team_access:
```

Comment:
Same here

Reply:
This is admittedly a half-way solution (in place of adding a two-level shared-accesss model), but in most cases (so we hear) Teams packages are usually shared to the whole team--and teams are frustrated that only one user can push--or completely private so this covers the missing common case.